mitiq/observable/pauli.py

```diff
@@ -146,10 +162,21 @@ def _expectation_from_measurements(
             measurements
         )
 
-    def __mul__(self, other: "PauliString") -> "PauliString":
-        result = PauliString()
-        result._pauli = self._pauli * other._pauli
-        return result
+    def __mul__(
+        self, other: Union["PauliString", complex, float, int]
+    ) -> "PauliString":
+        if isinstance(other, PauliString):
+            result = PauliString()
+            result._pauli = self._pauli * other._pauli
+            return result
```

Comment:
Same comment as before:
Here the private variable`result._pauli` is changed but the `__init__` method of the `PauliString` object is not called.
Is this safe? If not, returning a new `PauliString` directly  initialized from the string associated to `self._pauli * other._pauli` may be necessary.

Reply:
all resolved, good catch I didn't pay enough attention to `__init__`